server/src/main/java/org/apache/cassandra/sidecar/server/MainModule.java

```diff
@@ -182,9 +194,63 @@ public Vertx vertx(SidecarConfiguration sidecarConfiguration, MetricRegistryFact
         return Vertx.vertx(vertxOptions);
     }
 
+    @Provides
+    @Singleton
+    public IdentityToRoleCache identityRoleCache(Vertx vertx,
+                                                 SidecarConfiguration sidecarConfiguration,
+                                                 SystemAuthDatabaseAccessor systemAuthDatabaseAccessor)
+    {
+        IdentityToRoleCache identityToRoleCache
+        = new IdentityToRoleCache(vertx,
+                                  sidecarConfiguration.accessControlConfiguration().permissionCacheConfiguration(),
+                                  systemAuthDatabaseAccessor);
+        return identityToRoleCache;
+    }
+
+    @Provides
+    @Singleton
+    public ChainAuthHandler chainAuthHandler(Vertx vertx,
+                                             SidecarConfiguration sidecarConfiguration,
+                                             IdentityToRoleCache identityToRoleCache)
+    {
+        ChainAuthHandler chainAuthHandler = ChainAuthHandler.any();
+        try
+        {
+            MutualTlsAuthenticatorConfiguration mTLSAuthenticatorConfig = sidecarConfiguration
+                                                                          .accessControlConfiguration()
+                                                                          .authenticatorsConfiguration()
+                                                                          .mTlsAuthenticatorConfiguration();
+            if (mTLSAuthenticatorConfig.enabled())
+            {
+                CertificateValidator certificateValidator = (CertificateValidator) Class.forName(mTLSAuthenticatorConfig.certificateValidator()).newInstance();
+                CertificateIdentityExtractor certificateIdentityExtractor;
+                if (mTLSAuthenticatorConfig.certificateIdentityExtractor().equalsIgnoreCase(CassandraIdentityExtractor.class.getName()))
+                {
+                    certificateIdentityExtractor
+                    = new CassandraIdentityExtractor(identityToRoleCache, sidecarConfiguration.accessControlConfiguration().adminIdentities());
+                }
+                else
+                {
+                    certificateIdentityExtractor = new SpiffeIdentityExtractor();
+                }
```

Comment:
nit: `CertificateIdentityExtractor` can be created the same way as `CertificateValidator`
Not a big concern. Feel free to ignore.